### ISSUE TITLE
The DNS filter is removing the @metadata field

### ISSUE THREAD

**Issue by gmoskovicz** [logstash-plugins/logstash-filter-dns#17]
I am using a simple configuration and reading from elasticsearch. Steps to reproduce:

```
POST testcase/test
{
  "ipaddress": "127.0.0.1"
}
```

```
input {
  elasticsearch {
    hosts    => "127.0.0.1:9200"
    index    => "testcase"
    docinfo  => true
  }
}

filter {
  mutate {
    add_field => { 
      "hostname" => "%{ipaddress}"
      "index" => "%{[@metadata][_index]}"
      "document_type" => "%{[@metadata][_type]}"
      "document_id" => "%{[@metadata][_id]}"
    }
  }
  dns {
    nameserver => "127.0.0.1"
    reverse => [ "hostname" ]
    action  => "replace"
  }
}

output {
  stdout { 
    codec  => rubydebug {
      metadata => true
    }
  }
}
```

Output:

```
{
        "ipaddress" => "127.0.0.1",
         "@version" => "1",
       "@timestamp" => "2016-02-05T17:15:43.881Z",
         "hostname" => "localhost",
            "index" => "testcase",
    "document_type" => "test",
      "document_id" => "1"
}
```

If i remove the DNS filter, i get the desired result:

```
{
        "ipaddress" => "127.0.0.1",
         "@version" => "1",
       "@timestamp" => "2016-02-05T17:44:21.851Z",
         "hostname" => "127.0.0.1",
            "index" => "testcase",
    "document_type" => "test",
      "document_id" => "1",
        "@metadata" => {
        "_index" => "testcase",
         "_type" => "test",
           "_id" => "1"
    }
}
```

Looks like the DNS filter is removing the metadata?

**Comment by gmoskovicz** [comment]
https://github.com/logstash-plugins/logstash-filter-dns/blob/master/lib/logstash/filters/dns.rb#L68

`new_event = event.clone` can this cause issues?

**Comment by ph** [comment]
yes this is the issue `LogStash::Event` has a clone implementation but doesn't clone the metadata.

https://github.com/elastic/logstash/blob/master/logstash-core-event/lib/logstash/event.rb#L94-L102
```
  # Create a deep-ish copy of this event.
  def clone
    copy = {}
    @data.each do |k,v|
      # TODO(sissel): Recurse if this is a hash/array?
      copy[k] = begin v.clone rescue v end
    end

    self.class.new(copy)
  end
```

**Comment by jsvd** [comment]
fixed by #18 , released in 2.1.0
